acir/src/circuit/opcodes.rs

```diff
@@ -36,6 +36,44 @@ pub enum Opcode {
     // RAM is required for acvm-backend-barretenberg as dynamic memory implementation in Barretenberg requires an initialization phase and can only handle constant values for operations.
     RAM(MemoryBlock),
     Brillig(Brillig),
+    /// Atomic operation on a memory block
+    MemoryOp {
+        block_id: BlockId,
+        op: MemOp,
+    },
+    MemoryInit {
+        block_id: BlockId,
+        init: Vec<Witness>,
+    },
+}
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum OpcodeEnum {
```

Comment:
Why is this needed?

Reply:
This is needed to indicate that an opcode is not supported inside a CompilerError. The Opcode has a too big size to be used directly.

Reply:
I have renamed it, following the later comment.